haigha/connections/rabbit_connection.py

```diff
@@ -30,6 +31,24 @@ def __init__(self, **kwargs):
         class_map.setdefault(85, RabbitConfirmClass)
         kwargs['class_map'] = class_map
 
+        # Indicate RabbitMQ-specific consumer_cancel_notify capability per
```

Comment:
Aaron, the `haigha.Connection` constructor accepts a `client_properties` kwarg that it then uses to update the data from `connection.LIBRARY_PROPERTIES` and makes the combined result available to the broker inside the `Connection.Start-Ok` method. Most of my new code in this constructor assumes that `client_properties` may be passed in by user's outside of the library, so it's being extra careful not to trample `client_properties`, just in case it was passed in.

If, instead, you intended `client_properties` to be passed in only from the subclasses that are internal to Haigha, such as `RabbitConnection`, then I would like to remove most of this new code and simply assume that `RabbitConnection` constructor can set `client_properties` to whatever it wants. Please advise.